internal/pkg/dir/dir_test.go

```diff
@@ -57,3 +59,56 @@ func TestInit(t *testing.T) {
 	}
 	checkPermissions(t, bar, 0755)
 }
+
+func TestCopy(t *testing.T) {
+	src := t.TempDir()
+	srcDirname := path.Base(src)
+
+	// Test that empty directories don't copy anything
+	dst := t.TempDir()
+	err := Copy(src, dst)
+	if err != nil {
+		t.Fatalf("Unable to copy empty dir: %v", err)
+	}
+
+	//rmdir will fail if the directory has anything at all
+	err = exec.Command("rmdir", path.Join(dst, srcDirname)).Run()
```

Comment:
Maybe `os.Remove` instead of read :)

Reply:
Depends on what this check is supposed to prove. It looks like it is just to prove that the directory is empty, taking advantage of the fact that `rmdir` fails for non-empty directories. That's why I suggested `ReadDir` instead of `Remove`.